tests/calculators/test_torsion_calculators.py

```diff
@@ -0,0 +1,47 @@
+from pytest import approx
+import stko
+import stk
+
+
+def test_torsion_matching():
+    """confirm torsions are appropriately mapped from constructed
+    molecules to building blocks
+    """
+
+    def test_polymer(bb1: stk.BuildingBlock, bb2: stk.BuildingBlock):
+        """attach bb1 to bb2 and test torsion mapping"""
+        polymer = stk.ConstructedMolecule(
+            stk.polymer.Linear(
+                building_blocks=(bb1, bb2),
+                repeating_unit="AB",
+                orientations=[0, 0],
+                num_repeating_units=1
+            )
+        )
+
+        tors_calculator = stko.MatchedTorsionCalculator()
+        tors_results = tors_calculator.get_results(polymer)
+
+        angles = stko.get_torsion_info_angles(
+            polymer, list(tors_results.get_torsion_infos())[0])
+        return angles[0] == approx(angles[1])
+
+    bb1 = stk.BuildingBlock('NCCNCCN', [stk.PrimaryAminoFactory()])
+    bb2 = stk.BuildingBlock('O=CCCC=O', [stk.AldehydeFactory()])
+    assert test_polymer(bb1, bb2)
+
+    # test case designed such that default rdkit torsion for
+    # constructed molecule spans over both building blocks, requiring
+    # stko to correctly clean this up
+    bb3 = stk.BuildingBlock('CCCC', [stk.SingleAtom(stk.C(1))])
+    bb4 = stk.BuildingBlock('C', [stk.SingleAtom(stk.C(0))])
+    assert test_polymer(bb3, bb4)
+
+    # test case for an exterior atom of a building block torsion being
+    # deleted in construction
+    functional_group = stk.GenericFunctionalGroup(
+        atoms=(stk.C(0), stk.C(1)),
+        bonders=(stk.C(1),), deleters=(stk.C(0),))
```

Comment:
Reformat brackets.

Reply:
Also, this will need to include H atoms in the deleters.

Reply:
So, I need some time to go through this review properly but can comment on this quickly enough.

I would agree with you - I think the solution would be to use the `BuildingBlock.init` interface, where you specify atoms and bonds manually as stk objects, and do not go through rdkit at all, right? 

I normally use pymol or avogadro. I think avogadro will be the easiest as it should maintain the atom ids and bond info. Or, actually, Ovito is a good option for maintaining the atom ids - although they may be shifted by 1.

Reply:
@andrewtarzia I believe the barrier to the BuildingBlock.init interface is having to specify the position matrix manually. It seems a more user friendly option for rapid prototyping would be to have the simplicity of passing in a smiles string but the option of waiting to deal with hydrogen until after construction.

Thanks for the visualization suggestions!

Reply:
@joshkamm I see what you mean. Maybe BuildingBlock -> BuildingBlock.to_rdkit_mol() -> rdkit.removeHs -> BuildingBlock.init_from_rdkit_mol() (or some variation) could be used to just get your skeleton atoms?

Definitely submit an issue to stk of what you think a good interface would be and Lukas and I can have a think of a better approach!